app/controllers/reviews_controller.rb

```diff
@@ -12,6 +12,12 @@ def index
 
     @reviews_map[nil] = @reviews.where(semester_id: nil).sort_by{|r| r.course.mnemonic_number}
 
+    average_prof_rating = 0
+    @reviews.each do |r|
+      average_prof_rating += r.overall
+    end
+    @avg_prof_rating = (average_prof_rating/@reviews.count).round(2)
```

Comment:
depending on the filter type, you might want to move the logic actually to the model, rather than the controller layer, going for skinny controllers and fat models rather than the other way around